lib/core/application/event/event_team_members_form_bloc/event_team_members_form_bloc.dart

```diff
@@ -0,0 +1,228 @@
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'event_team_members_form_bloc.freezed.dart';
+
+class EventTeamMembersFormBloc
+    extends Bloc<EventTeamMembersFormBlocEvent, EventTeamMembersFormBlocState> {
+  final EventUserRole? initialEventUserRole;
+  EventTeamMembersFormBloc({this.initialEventUserRole})
+      : super(
+          EventTeamMembersFormBlocState.initial(),
+        ) {
+    on<EventTeamMembersFormBlocEventSelectRole>(onSelectRole);
+    on<EventTeamMembersFormBlocEventChangeVisibleOnEvent>(
+      onChangeVisibleOnEvent,
+    );
+    on<EventTeamMembersFormBlocEventAddNewUser>(
+      onAddNewUser,
+    );
+    on<EventTeamMembersFormBlocEventAddNewEmail>(
+      onAddNewEmail,
+    );
+    on<EventTeamMembersFormBlocEventRemoveUser>(
+      onRemoveUser,
+    );
+    on<EventTeamMembersFormBlocEventSubmitForm>(
+      onSubmitForm,
+    );
+    on<EventTeamMembersFormBlocEventReset>(
+      onReset,
+    );
+    on<EventTeamMembersFormBlocEventPopulateInitialUserRole>(
+      onPopulateUserRole,
+    );
+    if (initialEventUserRole != null) {
+      add(EventTeamMembersFormBlocEvent.populateInitialUserRole());
+    }
+  }
+
+  void onSelectRole(
+    EventTeamMembersFormBlocEventSelectRole event,
+    Emitter emit,
+  ) {
+    final isCohost = event.role?.code == Enum$RoleCode.Cohost;
+    final newState = state.copyWith(
+      selectedRole: event.role,
+      visibleOnEvent: isCohost == true ? true : false,
+    );
+    emit(
+      _validate(newState),
+    );
+  }
+
+  void onChangeVisibleOnEvent(
+    EventTeamMembersFormBlocEventChangeVisibleOnEvent event,
+    Emitter emit,
+  ) {
+    final newState = state.copyWith(
+      visibleOnEvent: event.visibleOnEvent,
+    );
+    emit(
+      _validate(newState),
+    );
+  }
+
+  void onAddNewUser(
+    EventTeamMembersFormBlocEventAddNewUser event,
+    Emitter emit,
+  ) {
+    List<dynamic> newUsers = [...state.users, event.user];
+
+    emit(
+      _validate(
+        state.copyWith(users: newUsers),
+      ),
+    );
+  }
+
+  void onAddNewEmail(
+    EventTeamMembersFormBlocEventAddNewEmail event,
+    Emitter emit,
+  ) {
+    List<dynamic> newUsers = [...state.users, event.email];
+
+    emit(
+      _validate(
+        state.copyWith(users: newUsers),
+      ),
+    );
+  }
+
+  void onRemoveUser(
+    EventTeamMembersFormBlocEventRemoveUser event,
+    Emitter emit,
+  ) {
+    List<dynamic> newUsers = [...state.users];
+    newUsers.removeAt(event.index);
+
+    emit(
+      _validate(
+        state.copyWith(users: newUsers),
+      ),
+    );
+  }
+
+  void onSubmitForm(
+    EventTeamMembersFormBlocEventSubmitForm event,
+    Emitter emit,
+  ) async {
+    emit(state.copyWith(status: EventTeamMembersFormStatus.loading));
+    final List<Input$UserFilter> usersInput = state.users
+        .map(
+          (item) => Input$UserFilter(
+            email: item is String ? item : null,
```

Comment:
should split into list of user, list of emails. 
Why use dynamic for both User and String ?